src/collect/collector.rs

```diff
@@ -249,6 +253,14 @@ impl Collectors {
         let mut json = JsonFormat::default();
         let mut processors = get_processors(&mut json, collect.args()?)?;
 
+        // Let the user know we got a termination signal and we're processing
```

Comment:
not sure if there's some limitation that prevents it, but have you considered replacing register_signal() with something similar to the code below?